vehicle-model-lifecycle/src/download_vspec.py

```diff
@@ -84,21 +87,104 @@ def download_file(uri: str, local_file_path: str):
                 outfile.write(chunk)
 
 
-def get_vehicle_model_src():
-    manifest_data_str = require_env("VELOCITAS_APP_MANIFEST")
-    manifest_data = json.loads(manifest_data_str)
+def is_legacy_app_manifest(app_manifest_dict: Dict[str, Any]) -> bool:
+    """Check if the used app manifest is a legacy file.
+
+    Args:
+        app_manifest_dict (Dict[str, Any]): The app manifest.
+
+    Returns:
+        bool: True if the app manifest is a legacy file, False if not.
+    """
+    return "manifestVersion" not in app_manifest_dict
+
+
+def get_legacy_model_src(app_manifest_dict: Dict[str, Any]) -> str:
+    """Get the source from the legacy vehicle model (app manifest < v3)
+
+    Args:
+        app_manifest_dict (Dict[str, Any]): The app manifest dict.
 
+    Returns:
+        str: The source URI of the vehicle model
+    """
     possible_keys = ["vehicleModel", "VehicleModel"]
```

Comment:
```suggestion
    possible_keys = ["vehicleModel", "VehicleModel", "vehiclemodel", "Vehiclemodel"]
```
maybe better with an regex? or de we have a rule somewhere for that?

Reply:
This is just the 2 different version from app manifest v1 and app manifest v2, any further variants are not supported. Hence I would not add these options or switch to regex.